### ISSUE TITLE
frameserver/vsignal - missing stepframe events

### ISSUE THREAD

**Issue by letoram** [letoram/arcan#160]
Reproduce:
1. take tests/counter, run inside durden
2. see if stepframe is emitted (color changes), if they do, input mouse events
   and see if it stops.

Hypotheses:
1. stepframes not being delivered on full event queue
2. event-queue race condition
3. synch-strategy interference where release-herd doesn't trigger stepframe

**Comment by letoram** [comment]
The 'event-queue' full situation needs to be handled both in the _frameserver
side in the engine, and in the shmif-server implementation.

Intended fix:
1. in the normal poll-stage, check if there is a deferred notification, and
   if so, re-attempt delivery

2. if the event-queue is continuously full, try and poke the semaphore to
   unblock an open client.

3. better squishing of high-rate events, and migrate to mmio:ed mouse state.

**Comment by letoram** [comment]
fixed with ca5a917
